dace/dtypes.py

```diff
@@ -68,6 +68,17 @@ class ReductionType(aenum.AutoNumberEnum):
     Max_Location = ()  # Maximum value and its location
 
 
+@extensible_enum
+class AllocationLifetime(aenum.AutoNumberEnum):
+    """ Options for allocation span (when to allocate/deallocate) of data. """
+
+    Scope = ()  # Allocated/Deallocated on innermost scope start/end
+    State = ()  # Allocated throughout the containing state
+    SDFG = ()  # Allocated throughout the innermost SDFG (possibly nested)
+    TopLevel_SDFG = ()  # Allocated throughout the entire program (outer SDFG)
```

Comment:
Maybe "Global"? Would be more intuitive for newcomers

Reply:
Sounds good, renamed 👍